### PR TITLE
Improve S1192 by ignoring literals in type annotations

### PR DESCRIPTION
**Pull request by yassin-kammoun-sonarsource**
Fixes [#2464](https://github.com/SonarSource/SonarJS/issues/2464)

### CODE REVIEW

File: tests/rules/no-duplicate-string.test.ts [hunk 1]
```diff
@@ -20,6 +20,7 @@
 import { RuleTester } from "eslint";
 
 const ruleTester = new RuleTester({
+  parser: require.resolve("@typescript-eslint/parser"),
```
Comment (saberduck) [inline tests/rules/no-duplicate-string.test.ts:23]: can we use `RuleTesterJsWithTypes ` from Andrea?

**Review by saberduck — APPROVED** [review]

File: tests/rules/no-duplicate-string.test.ts (hunk 1 above)
Comment (yassin-kammoun-sonarsource) [reply tests/rules/no-duplicate-string.test.ts:23]: Well, the class is defined in SonarJS, therefore not available in eslint-plugin-sonarjs. However, we could temporarily duplicate its implementation here in eslint-plugin-sonarjs, create a ticket to remove the one in SonarJS, and eventually use eslint-plugin-sonarjs's one once we release it.

What do you think ?

Comment (yassin-kammoun-sonarsource) [reply tests/rules/no-duplicate-string.test.ts:23]: @saberduck ?

Comment (saberduck) [reply tests/rules/no-duplicate-string.test.ts:23]: yep, that could be a good idea, however maybe in separate PR, I would merge this as is

**Pull request merged**